src/frontend/src/reducers/networks/networksThunk.ts

```diff
@@ -54,65 +57,105 @@ export const getNetworks = createAsyncThunk(
+                                timesFetchPromise,
+                            ]);
 
-                    const pathsData = pathsResponse
-                        ? parsePathsData(pathsResponse.value)
-                        : ({} as ParsedPathsData);
-                    const travelTimeSurface = parseTimesData(
-                        timesResponse.value
-                    );
+                        const pathsData = pathsResponse
+                            ? parsePathsData(pathsResponse.value)
+                            : ({} as ParsedPathsData);
 
-                    return {
-                        name: network as NetworkModeOptionKey,
-                        ...pathsData,
-                        travelTimeSurface: travelTimeSurface,
-                        timesAndPathsDataReady: true,
-                    };
-                })
-            );
-            return allParsedTimeAndPathData.reduce(
-                (dataByNetwork, data) => {
-                    dataByNetwork[data.name] = data;
-                    return dataByNetwork;
-                },
-                {} as { [key in NetworkModeOptionKey]: TimesAndPathsData }
-            );
-        } catch (error: any) {
-            return rejectWithValue(error.message);
+                        const travelTimeSurface = parseTimesData(
+                            timesResponse.value
+                        );
+
+                        const routableNetwork = {
+                            ...networkDetails,
+                            name: network as NetworkModeOptionKey,
+                            ...pathsData,
+                            travelTimeSurface: travelTimeSurface,
+                        } as NetworkAndTimeAndPathsData;
+
+                        // derive neighborhood routes by place and network mode
+                        const networkRoutesByNeighborhood =
+                            createNetworkNeighborhoodRoutes(
+                                routableNetwork,
+                                origin,
+                                neighborhoods
+                            );
+                        // get travel times by place and network mode
+                        const networkTravelTimesByNeighborhood =
+                            createNetworkNeighborhoodTravelTimes(
+                                routableNetwork,
+                                neighborhoods
+                            );
+
+                        return {
+                            name: network as NetworkModeOptionKey,
+                            routesByNeighborhood: networkRoutesByNeighborhood,
+                            travelTimesByNeighborhood:
+                                networkTravelTimesByNeighborhood,
+                            timesAndRoutesDataReady: true,
+                        };
+                    })
+                );
+                return {
+                    label: label,
```

Comment:
After we have the user sign-in and profile hooked up with the backend API, I'd suggest we use the destination record's ID instead of the label (currently name or address here). If we are going with this solution, I'd suggest we add a TODO here.

Reply:
Added this comment in c670861f359d18f80e79031c982e2f2391baedc4